pentaho-js/src/main/javascript/cdf/components/AnalyzerComponent.js

```diff
@@ -43,19 +44,27 @@ define([
     },
 
     getOptions: function() {
+      var myself = this;
       var options = {
         solution: this.solution,
         path: this.path,
         action: this.action,
         command: this.command == undefined ? "open" : this.command,
         showFieldList: this.showFieldList == undefined ? false : this.showFieldList,
         showRepositoryButtons: this.showRepositoryButtons == undefined ? false : this.showRepositoryButtons,
-        frameless: this.frameless == undefined ? false : this.frameless
+        frameless: this.frameless == undefined ? false : this.frameless,
+        dateFormats: this.dateFormats == undefined ? {} : this.dateFormats
```

Comment:
Don't pass dateFormats to the **options** object, it will be used to build the url and that's not what we want here.
@afrjorge made me aware of the legacy complications, so removing this the object and adding it in below should suffice. (`this.dateFormats == undefined ? {} : this.dateFormats` for example)